tajo-core/src/main/java/org/apache/tajo/master/QueryManager.java

```diff
@@ -121,19 +126,27 @@ public EventHandler getEventHandler() {
 
   public synchronized Collection<QueryInfo> getFinishedQueries() {
```

Comment:
In some cases, it may result in lots of histories. In that case, It will cause long latency. Later, we need to improve the paging feature in new jira.